invisible_cities/cities/components.py

```diff
@@ -1002,3 +1033,276 @@ def calculate_and_save_buffers(buffer_length    : float        ,
+                ave_r   = np.average(r  , weights=e, axis=0)
+                extr1, extr2 = plf.find_extrema(t)
+                extr1_pos = extr1.XYZ
+                extr2_pos = extr2.XYZ
+
+                blob_pos1, blob_pos2 = plf.blob_centres(t, blob_radius)
+
+                e_blob1, e_blob2, hits_blob1, hits_blob2 = plf.blob_energies_and_hits(t, blob_radius)
+                overlap = float(sum(h.Ep for h in set(hits_blob1).intersection(set(hits_blob2))))
+                list_of_vars = [hitc.event, tID, energy, length, numb_of_voxels,
+                                numb_of_hits, numb_of_tracks,
+                                min(x), min(y), min(z), min(r), max(x), max(y), max(z), max(r),
+                                *ave_pos, ave_r, *extr1_pos,
+                                *extr2_pos, *blob_pos1, *blob_pos2,
+                                e_blob1, e_blob2, overlap,
+                                vox_size_x, vox_size_y, vox_size_z]
+
+                df.loc[c] = list_of_vars
+
+                for vox in t.nodes():
+                    for hit in vox.hits:
+                        hit.track_id = tID
+                        track_hits.append(hit)
+
+            #change dtype of columns to match type of variables
+            df = df.apply(lambda x : x.astype(types_dict_tracks[x.name]))
+            track_hitc.hits.extend(track_hits)
+        return df, track_hitc, out_of_map
+
+    return create_extract_track_blob_info
+
+
+def compute_and_write_tracks_info(paolina_params, h5out, hits_type=None):
+
+    # Create tracks and compute topology-related information
+    create_extract_track_blob_info = fl.map(track_blob_info_creator_extractor(**paolina_params),
+                                            args = 'Ep_hits',
+                                            out  = ('topology_info', 'paolina_hits', 'out_of_map'))
+
+    # Filter empty topology events
+    filter_events_topology         = fl.map(lambda x : len(x) > 0,
+                                            args = 'topology_info',
+                                            out  = 'topology_passed')
+    events_passed_topology         = fl.count_filter(bool, args="topology_passed")
+
+    # Create table with summary information
+    make_final_summary             = fl.map(make_event_summary,
+                                            args = ('event_number', 'topology_info', 'paolina_hits', 'out_of_map'),
+                                            out  = 'event_info')
+
+
+    # Define writers and make them sinks
+    write_tracks          = fl.sink(   track_writer     (h5out=h5out)             , args="topology_info"      )
+    write_summary         = fl.sink( summary_writer     (h5out=h5out)             , args="event_info"         )
+    write_topology_filter = fl.sink( event_filter_writer(h5out, "topology_select"), args=("event_number", "topology_passed"    ))
+
+    if hits_type:
+        group_table = hits_type.split("/")[0]
```

Comment:
As we have discussed offline, it doesn't make a lot of sense the possibility to save the deconvoluted hits, therefore I remove the option for allowing to change the name of the table. Now there is just a flag called `save_paolina_hits` that will be True, in case the function is executed inside Esmeralda (saving the hits in the table `CHITS/highTh`) and it will correspond to False, if the function is run inside Isaura.